.github/workflows/nix.yml

```
@@ -0,0 +1,23 @@
+name: CI - Nix
+
+on: [push, pull_request]
```

```suggestion
on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main
```

otherwise, with this syntax, when you push a commit to an open PR, you'll get duplicate CI jobs